src/pgwire/src/codec.rs

```diff
@@ -463,7 +463,7 @@ where
         VERSION_SSL => FrontendStartupMessage::SslRequest,
         VERSION_GSSENC => FrontendStartupMessage::GssEncRequest,
         _ => {
-            let mut params = HashMap::new();
+            let mut params = BTreeMap::new();
```

Comment:
Looking through the code, it doesn't seem like this ever gets that big, and we do in fact iterate over this map eventually. So it's probably for the best to change it to a `BTreeMap`, but do you happen to know if any of the benchmarks happen to test connection time? @philip-stoev do you know the answer to this?